newdle/client/src/components/summary/SummaryPage.js

```diff
@@ -193,9 +229,35 @@ export default function SummaryPage() {
                 </div>
               )}
             </ParticipantTable>
-          </div>
+          )}
           {isCreator && (
             <div className={styles['button-row']}>
+              {gridViewActive && (
+                <>
+                  {hasParticipantsWithEmail && (
+                    <Button
+                      icon
+                      color="blue"
+                      labelPosition="left"
+                      loading={mailSending}
+                      disabled={mailSending || mailSent}
+                      onClick={() => setMailModalOpen(true)}
+                    >
+                      <Icon name="mail" />
+                      <Trans>E-mail participants</Trans>
+                    </Button>
+                  )}
+                  {config.has_event_creation && (
+                    <Button
+                      className={styles['create-event-button']}
+                      onClick={createEvent}
+                      loading={loading}
+                    >
+                      <Trans>Create event</Trans>
+                    </Button>
+                  )}
+                </>
```

Comment:
We have this exact same code twice in that file now. Can we avoid this duplication?

Reply:
Moving the duplicate above the `return` block should be enough:

```jsx
  const actions = (
    <>
      {hasParticipantsWithEmail && (
        <Button
          icon
          color="blue"
          labelPosition="left"
          loading={mailSending}
          disabled={mailSending || mailSent}
          onClick={() => setMailModalOpen(true)}
        >
          <Icon name="mail" />
          <Trans>E-mail participants</Trans>
        </Button>
      )}
      {config.has_event_creation && (
        <Button className={styles['create-event-button']} onClick={createEvent} loading={loading}>
          <Trans>Create event</Trans>
        </Button>
      )}
    </>
  );
```

Reply:
Thanks, I'll update it :+1: